api/tests/integration/edge_api/identities/test_edge_identity_featurestates_viewset.py

```diff
@@ -559,6 +552,167 @@ def test_edge_identities_post_returns_400_for_invalid_mvfs_allocation(
+    assert actual_feature_state["multivariate_feature_state_values"] == []
+    assert actual_feature_state["featurestate_uuid"] is not None
+
+
+@pytest.mark.parametrize(
+    "lazy_feature", [(lazy_fixture("feature")), (lazy_fixture("feature_name"))]
+)
+def test_edge_identities_with_identifier_delete_featurestate(
+    admin_client,
+    environment,
+    environment_api_key,
+    identity_document,
+    dynamo_wrapper_mock,
+    feature,
+    lazy_feature,
+):
+    # Given
+    dynamo_wrapper_mock.get_item.return_value = identity_document
+    identifier = identity_document["identifier"]
+    url = reverse(
+        "api-v1:environments:edge-identities-with-identifier-featurestates",
+        args=[environment_api_key],
+    )
+    data = {"identifier": identifier, "feature": lazy_feature}
+
+    # When
+    response = admin_client.delete(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    dynamo_wrapper_mock.get_item.assert_called_with(
+        f"{environment_api_key}_{identifier}"
+    )
+
+    # Next, let's verify that deleted feature state
+    # is not part of identity dict that we put
+    name, args, _ = dynamo_wrapper_mock.mock_calls[1]
+    assert name == "put_item"
+    assert not list(
+        filter(
+            lambda fs: fs["feature"]["id"] == feature,
+            args[0]["identity_features"],
+        )
+    )
+    assert response.status_code == status.HTTP_204_NO_CONTENT
+
+
+def test_edge_identities_with_identifier_update_featurestate(
+    admin_client,
+    environment,
+    environment_api_key,
+    identity_document,
+    dynamo_wrapper_mock,
+    feature,
+):
+    # Given
+    dynamo_wrapper_mock.get_item.return_value = identity_document
+    identifier = identity_document["identifier"]
+    url = reverse(
+        "api-v1:environments:edge-identities-with-identifier-featurestates",
+        args=[environment_api_key],
+    )
+    expected_feature_state_value = "new_feature_state_value"
+    expected_fs_enabled = True
+    data = {
+        "multivariate_feature_state_values": [],
+        "enabled": expected_fs_enabled,
+        "feature": feature,
+        "feature_state_value": expected_feature_state_value,
+        "identifier": identifier,
+    }
+
+    # When
+    response = admin_client.put(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+    # Then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["feature"] == feature
+    assert response.json()["feature_state_value"] == expected_feature_state_value
+    assert response.json()["enabled"] == data["enabled"]
+    dynamo_wrapper_mock.get_item.assert_called_with(
+        f"{environment_api_key}_{identifier}"
+    )
+    name, args, _ = dynamo_wrapper_mock.mock_calls[1]
+    assert name == "put_item"
+
+    # Next, let's verify that the document that we put
+    # have correct updates
+
+    # First, let's modify the fs value that we updated
```

Comment:
I think it would be good to do this on a copy of the dictionary, right? I'm not exactly sure how this will work if we're modifying the object that we specified should be returned in the logic flow. It could be (although I don't think so) that you're modifying the same object in python memory that you're then asserting is equal to it later on in this test. 

```python
identity_document_copy = copy.deepcopy(identity_document)
identity_document_copy["identity_features"][0]["feature_state_value"] = expected_feature_state_value
...
```

Reply:
No, it's actually not that's what I thought as well, but the dictionary that we receive from `args[0]` is a completely different object that flag-engine generated from a model

Reply:
Ok, cool.